spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/utils/ClassloaderUtils.java

```diff
@@ -50,16 +53,23 @@ public static ClassLoader createModuleClassloader(URL[] urls) {
 			// to compensate for LaunchedURLClassLoader not delegating to parent to retrieve resources
 			@SuppressWarnings("resource")
 			URLClassLoader systemUrlClassLoader = (URLClassLoader) systemClassLoader;
-			URL[] mergedUrls = new URL[urls.length + systemUrlClassLoader.getURLs().length];
+			List<URL> systemURLs = new ArrayList<>();
+			for (URL url: systemUrlClassLoader.getURLs()) {
+				// this will help avoiding any tests using the classes directory in the system class loader.
+				if (!url.getFile().endsWith("classes/")) {
```

Comment:
Can we perhaps control this through a configuration parameter/method argument - maybe with an override that preserves the default behaviour? Not worth investing much time in the configuration, but I don't think we should do exclusions based on a hardcoded path.